src/replay/scheduler.zig

```diff
@@ -1,106 +1,233 @@
+//! Helper Structs and functions used in execution.zig when asynchronously
+//! executing slots
+
 const std = @import("std");
 const sig = @import("../sig.zig");
 const replay = @import("lib.zig");
 const tracy = @import("tracy");
 
+const core = sig.core;
+
 const Allocator = std.mem.Allocator;
 const Atomic = std.atomic.Value;
 
 const Channel = sig.sync.Channel;
 const HomogeneousThreadPool = sig.utils.thread.HomogeneousThreadPool;
 const ThreadPool = sig.sync.ThreadPool;
 
-const Hash = sig.core.Hash;
+const Entry = core.Entry;
+const Hash = core.Hash;
 const Transaction = sig.core.Transaction;
 
+const ParsedVote = sig.consensus.vote_listener.vote_parser.ParsedVote;
+const SlotHashes = sig.runtime.sysvar.SlotHashes;
 const TransactionError = sig.ledger.transaction_status.TransactionError;
 
 const AccountLocks = replay.account_locks.AccountLocks;
+const BatchResult = replay.execution.BatchResult;
 const Committer = replay.commit.Committer;
-const ConfirmSlotError = replay.confirm_slot.ConfirmSlotError;
-const ConfirmSlotStatus = replay.confirm_slot.ConfirmSlotStatus;
-const ResolvedTransaction = replay.resolve_lookup.ResolvedTransaction;
+const ReplaySlotError = replay.execution.ReplaySlotError;
 const ResolvedBatch = replay.resolve_lookup.ResolvedBatch;
+const ResolvedTransaction = replay.resolve_lookup.ResolvedTransaction;
+const SlotResolver = replay.resolve_lookup.SlotResolver;
 const SvmGateway = replay.svm_gateway.SvmGateway;
 
-const ProcessedTransaction = sig.runtime.transaction_execution.ProcessedTransaction;
-
-const executeTransaction = replay.svm_gateway.executeTransaction;
-const preprocessTransaction = replay.preprocess_transaction.preprocessTransaction;
-
-const Logger = sig.trace.Logger("replay-batcher");
-
-const vote_listener = sig.consensus.vote_listener;
-const ParsedVote = vote_listener.vote_parser.ParsedVote;
+const verifyPoh = core.entry.verifyPoh;
+const resolveBatch = replay.resolve_lookup.resolveBatch;
 
 const assert = std.debug.assert;
 
-/// Processes a batch of transactions by verifying their signatures and
-/// executing them with the SVM.
-pub fn processBatch(
+const Logger = sig.trace.Logger("replay-async");
+
+/// schedule poh verification asynchronously
+pub fn startPohVerify(
     allocator: Allocator,
-    svm_params: SvmGateway.Params,
-    committer: Committer,
-    transactions: []const ResolvedTransaction,
+    logger: Logger,
+    pool: *HomogeneousThreadPool(PohTask),
+    initial_hash: Hash,
+    entries: []const Entry,
     exit: *Atomic(bool),
-) !BatchResult {
-    var zone = tracy.Zone.init(@src(), .{ .name = "processBatch" });
-    zone.value(transactions.len);
```

Comment:
Is the removal of the tracy zone here intentional? Or is this just a weird diff?

Reply:
This is a different function. `processBatch` was moved to `execution.zig`.

Reply:
Weird batch, gotcha